src/game_api/level_api.hpp

```diff
@@ -169,80 +176,91 @@ class ThemeInfo
     // jungle: when black market present
     // volcana: when drill present
     // touches the rooms and sometimes the meta info about the rooms
-    virtual void handle_level_specialities() = 0;
+    // ???
+    virtual void add_special_rooms() = 0;
 
     // can't trigger, dwelling (quillback) and abzu do something special (arena just returns)
     virtual void unknown_v7() = 0;
 
     // does something depending on levelgen.data.unknown7
     virtual void unknown_v8() = 0;
 
-    virtual void insert_shopkeeper_vault() = 0;
+    // ???
+    virtual void add_vault() = 0;
 
-    virtual void insert_coffin() = 0;
+    // ???
+    virtual void add_coffin() = 0;
 
     // metal clanking and air of oppression
-    virtual void set_theme_specific_level_feeling() = 0;
+    virtual void add_special_feeling() = 0;
 
-    // Note: Inserted somewhere between initialize_flags and populate_level
+    // Note: Inserted somewhere between init_flags and spawn_level
     virtual bool unknown_v12() = 0;
 
+    // spawns all floor etc tiles based on the room layout
     // disable this and only the player is spawned in the level
-    virtual void populate_level() = 0;
+    virtual void spawn_level() = 0;
 
-    virtual void add_level_bordertiles() = 0;
+    // spawns CO: teleportingborder / Duat: dust / theme specific border tiles around the level
+    virtual void spawn_border() = 0;
 
     // volcana: checks if state.coffin_contents = cocovondiamonds -> chooses one of the four coffins in vlad's castle at random to put her in
     // tidepool: spawns impostor lake, some door and a litwalltorch, ...
     // neobab: assigns the correct animation_frame to all the ushabtis
     // co, arena: forwards to the same virtual of the sub_theme
     // abzu, tiamat: spawns impostor lake
+    // ???
     virtual void post_process_level() = 0;
 
+    // adds theme specific random traps and pushblocks
     // quillback: adds butterflies and snails + configures the door to go to the correct theme/level
```

Comment:
I assume this comment about butterflies is wrong now?

Reply:
Maybe, I don't know if the virtuals got shuffled or changed or not tested correctly, but none of the old comments are to be trusted.